build_tools/bazel_to_cmake/bazel_to_cmake.py

```diff
@@ -297,33 +299,26 @@ def glob(self, include, exclude=[], exclude_directories=1):
     # See https://cmake.org/cmake/help/v3.12/command/file.html#filesystem
 
     if exclude_directories != 1:
-      raise NotImplementedError("Non-default exclude_directories not supported")
+      self._convert_unimplemented_function("glob", "with exclude")
+    if exclude:
+      self._convert_unimplemented_function("glob", "with exclude_directories")
 
-    filepaths = []
+    glob_vars = []
     for pattern in include:
       if "**" in pattern:
         # bazel's glob has some specific restrictions about crossing package boundaries.
         # We have no uses of recursive globs. Rather than try to emulate them or
         # silently give different behavior, just error out.
         # See https://docs.bazel.build/versions/master/be/functions.html#glob
         raise NotImplementedError("Recursive globs not supported")
-
-      filepaths += glob.glob(self.converter.directory_path + "/" + pattern)
-
-    exclude_filepaths = set([])
-    for pattern in exclude:
-      if "**" in pattern:
-        # See comment above
-        raise NotImplementedError("Recursive globs not supported")
-      exclude_filepaths.update(
-          glob.glob(self.converter.directory_path + "/" + pattern))
-
-    basenames = sorted([
-        os.path.basename(path)
-        for path in filepaths
-        if path not in exclude_filepaths
-    ])
-    return basenames
+      glob_var = "_GLOB_" + pattern.replace("*", "X").replace(".", "_").upper()
```

Comment:
As in some other places within this file, want to include a sample input and output for this replacement? There are lots of different conversions happening throughout this file, so it helps to explain at each step what should be happening.

```
# Bazel `*.mlir` glob -> CMake Variable `_GLOB_X_MLIR`
```

Example:
https://github.com/google/iree/blob/d7dd75d137294daaf4ceda718a464dbd4e7b45a9/build_tools/bazel_to_cmake/bazel_to_cmake.py#L128-L134

(Refactoring into little helper functions and writing unit tests would also serve this purpose)

Reply:
Done (the first thing, not the second thing :-P)